### ISSUE TITLE
`prefer_int_literals` Lint rule does not apply to constructor default values for `double`

### ISSUE THREAD

**Issue by KKimj** [dart-lang/dart_style#1670]
```dart
// prefer_int_literals

class A {
  const A({
    // Expected behavior: Lint warning should appear for 0.0, but actual behavior: No lint warning.
    this.bad = 0.0,
    this.good = 0,
  });

  final double bad;
  final double good;
}
```

In the following code, the `prefer_int_literals` lint rule does not trigger when `0.0` is used as the default value for a `double type` parameter, even though it should. 

## Expected Behavior:
Linter should show a warning when `0.0` is used as the default value for a `double type` parameter.

## Actual Behavior:
The lint rule doesn't show any warning for `0.0` used as a default value for `bad`.

**Comment by KKimj** [comment]
Dup w/ https://github.com/dart-lang/sdk/issues/58490,
